internal/config/config_test.go

```diff
@@ -601,6 +659,99 @@ address-pools:
   - communities: ["flarb"]
 `,
 		},
+
+		{
```

Comment:
Adding comment on a random line, as I can't comment on the file (I miss git send-email :)).

Following my comment on the `validatePeerAutodiscovery()`, if peers with:
* All fields fromAnnotations valid except one
 * That invalid field in `fromAnnotations` is valid in the `fromLabels` equivalent

is a valid peer, please test those cases too. Those edge cases seem very important, IMHO, to check the behavior is actually what we expect.

Reply:
The way I see it, an **invalid** value specified by the user should always result in an error (even if there is a valid value in a lower-precedence entity). For example, if the user specifies `oops` as the local ASN in an annotation and also specifies `65001` as the local ASN in a label, autodiscovery will fail with an error because `oops` isn't a valid ASN. IMO this is the desired behavior, i.e. I'd rather let the user know their config is wrong than try to continue "at all costs" to find a valid BGP parameter somewhere.

I've added a few unit tests to explicitly test invalid values as well as the case where invalid values are specified in annotations while valid values are specified in labels.

Reply:
Thanks!